EuclideanGeometry/Foundation/Construction/Triangle/Orthocenter.lean

```diff
@@ -10,7 +11,15 @@ variable {P : Type _} [EuclideanPlane P]
 
 def TriangleND.Height1 (tr_nd : TriangleND P) : SegND P := sorry
 
-structure IsOrthocenter (tr_nd : TriangleND P) (H : P) : Prop where
+-- this takes care of corner cases such as right triangles, where AH ⟂ BC runs into trouble since A = H
+def IsOrthocenter (tr_nd : TriangleND P) (H : P) : Prop := inner (VEC tr_nd.point₁ H) (VEC tr_nd.point₂ tr_nd.point₃) = (0 : ℝ) ∧ inner (VEC tr_nd.point₂ H) (VEC tr_nd.point₃ tr_nd.point₁) = (0 : ℝ) ∧ inner (VEC tr_nd.point₃ H) (VEC tr_nd.point₁ tr_nd.point₂) = (0 : ℝ)
+
+theorem orthocenter_exists (tr : Triangle P) (H : P) (h₁ : inner (VEC tr.point₁ H) (VEC tr.point₂ tr.point₃) = (0 : ℝ)) (h₂ : inner (VEC tr.point₂ H) (VEC tr.point₃ tr.point₁) = (0 : ℝ)) : inner (VEC tr.point₃ H) (VEC tr.point₁ tr.point₂) = (0 : ℝ) := by
```

Comment:
This theorem could be instead stated as a `def` that takes in an triangle and output a point. Then check this def do satisfies `IsOrthocenter`

Reply:
This strategy is used in the file `AngBisector`. Maybe first writing a plan file or discussing directly with me is more time-saving!

Reply:
Well, this can be implemented.  But surely there is no need to complete everything in one commit?